openedx_learning/apps/authoring/publishing/model_mixins.py

```diff
@@ -28,17 +30,12 @@ class PublishableEntityMixin(models.Model):
     the publishing app's api.register_content_models (see its docstring for
     details).
     """
-
-    class PublishableEntityMixinManager(models.Manager):
-        def get_queryset(self) -> QuerySet:
-            return super().get_queryset() \
-                          .select_related(
-                              "publishable_entity",
-                              "publishable_entity__published",
-                              "publishable_entity__draft",
-                          )
-
-    objects: models.Manager[PublishableEntityMixin] = PublishableEntityMixinManager()
+    # select these related entities by default for all queries
+    objects: ClassVar[WithRelationsManager[Self]] = WithRelationsManager(
```

Comment:
It's actually possible to remove this `ClassVar[WithRelationsManager[Self]]` annotation, and everything will still be correct. Though it seems VS Code / Pyright then interprets it as `Manager[Self]` rather than `WithRelationsManager[Self]`, whereas mypy is unaffected. I don't have a strong opinion but it seems fine to be more explicit.